### DISCUSSION TITLE
[mini.align] Align by treesitter nodes

### DISCUSSION THREAD

**Discussion by roginfarrer** [nvim-mini/mini.nvim#1442] [Q&A] [answered]
I think it would be great if there would be a way to align text by treesitter nodes. Maybe that's out of scope for this plugin because it's so regex centric, but there's no plugin that provides this feature as far as I know.

My use case is formatting the columns of the keys list in a lazy.nvim spec. Typical spec looks like:

```
{ '<lhs>', '<rhs>', ...opts }
```

I think it would be awesome to be able to separate those columns by the treesitter node. Otherwise you need to get really creative with the separators. A `,` would work great unless the `<rhs>` is a function that includes a column. Maybe it's possible to provide a separator lua pattern that ensures it's a comma that's not between parens, but I haven't had much luck with it.

**Comment by echasnovski** [comment, marked as answer]
Thanks for the suggestion, but that is indeed out of scope for 'mini.align'. The best approximation is to use [filter step](https://github.com/echasnovski/mini.nvim/blob/733afb94ed88d32b839a644dc137604691e46032/doc/mini-align.txt#L260) which has [several special variables](https://github.com/echasnovski/mini.nvim/blob/733afb94ed88d32b839a644dc137604691e46032/doc/mini-align.txt#L841-L850) available for easier interactive filtering.
